src/buy/components/BuyButton.tsx

```diff
@@ -0,0 +1,68 @@
+import { useCallback, useMemo } from 'react';
+import { Spinner } from '../../internal/components/Spinner';
+import { checkmarkSvg } from '../../internal/svg/checkmarkSvg';
+import {
+  background,
+  border,
+  cn,
+  color,
+  pressable,
+  text,
+} from '../../styles/theme';
+import { useBuyContext } from './BuyProvider';
+
+export function BuyButton() {
+  const {
+    setIsDropdownOpen,
+    from,
+    fromETH,
+    fromUSDC,
+    to,
+    lifecycleStatus: { statusName },
+  } = useBuyContext();
+  const isLoading =
+    to?.loading ||
+    from?.loading ||
+    fromETH?.loading ||
+    fromUSDC?.loading ||
+    statusName === 'transactionPending' ||
+    statusName === 'transactionApproved';
+
+  const isDisabled = !to?.amount || !to?.token || isLoading;
+
+  const handleSubmit = useCallback(() => {
+    setIsDropdownOpen(true);
+  }, [setIsDropdownOpen]);
+
+  const buttonContent = useMemo(() => {
+    if (statusName === 'success') {
+      return checkmarkSvg;
```

Comment:
should this be using the same icons as transactiontoast and swaptoast?  https://github.com/coinbase/onchainkit/blob/main/src/transaction/components/TransactionToastIcon.tsx